### PR TITLE
Informative note about role of id attribute

### LINKED ISSUE #911: Passing a transaction id to a payment request
My understanding is a typical payment integration works like this:

0. A merchant server holds a key and a secret from the payment handler.
1. As a customer proceeds to checkout, the merchant sends a request to the backend.
2. The merchant server sends a signed request to the payment handler with the total price, reserves a transaction id and receives a payment URL to redirect the customer to.
3. The merchant redirects the customer to the payment URL on the payment handler website.
4. As soon as the customer finishes the payment at the payment handler website, they are redirected back to the merchant with the resulting credential and the transaction id in query.
5. The merchant determines the user with the transaction id (or with a session id?)
6. The merchant verifies the payment by sending the transaction id and the resulting credential.
7. Successful purchase.

If we apply a similar scenario to the Payment Request API, the key would be how they pass the transaction id to the PR API.

**Question**: is `id` in the [`PaymentDetailsInit`](https://w3c.github.io/payment-request/#paymentdetailsinit-dictionary) the property a merchant can use to specify a transaction id?
I thought of `data` property of [`PaymentMethodData`](https://w3c.github.io/payment-request/#paymentmethoddata-dictionary), but it has to be constructed before displaying the payment button (preferably) and the total price may change at the point the customer press the button. Hence being able to pass a transaction id in the `detailsPromise` after the `.show()` is called would be efficient for merchants IMO.

Let me know what is the best solution here.

### PR DESCRIPTION
**Pull request by ianbjacobs**
closes #911 

The following tasks have been completed:

 * [x] Confirmed there are no ReSpec errors/warnings.
 * [ ] Has undergone security/privacy review (link)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/pull/913.html" title="Last updated on May 7, 2020, 12:38 PM UTC (057492c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/913/f868c93...057492c.html" title="Last updated on May 7, 2020, 12:38 PM UTC (057492c)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -917,6 +917,14 @@ <h2>
           {{PaymentRequest}}'s
           [=PaymentRequest/[[details]]=].{{PaymentDetailsInit/id}}.
         </p>
+        <div class="note">
+          <p>
+            For auditing and reconciliation purposes, a merchant can assign a
+            unique identifier for each transaction with the <a>id</a>
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
            unique identifier for each transaction with the {{PaymentDetailsInit/id}}
```

**Comment by marcoscaceres** [comment]
Seems a little bit redundant to add this, but 🤷‍♂️.

**Comment by ianbjacobs** [comment]
@marcoscaceres, I simplified the note to the one sentence about the purpose of the attribute (which is not stated elsewhere as far as I can tell). I removed the bit about browser-generated ID when not provided by the merchant; that is covered in the algorithm.

**Review by marcoscaceres — APPROVED** [review]
Lgtm

**Comment by ianbjacobs** [comment]
(Wondering why checks haven't completed...)

**Comment by marcoscaceres** [comment]
Used my super powers.

**Pull request merged**